### PR TITLE
fix: indexing dev docs by splitting the docs build stage.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
Fix the docs indexing error

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -145,9 +145,15 @@ jobs:
           cname: ${{ env.DOCUMENTATION_CNAME }}
           token: ${{ secrets.GITHUB_TOKEN }}
 
+
+  doc-index-dev:
+    name: "Deploy dev docs index"
+    #if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: docs-build
```
Comment (RobPasMue) [inline .github/workflows/ci-build.yml:153]: This should run after deploying the docs @Revathyvenugopal162 - please put a ``needs`` stage

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (Revathyvenugopal162) [reply .github/workflows/ci-build.yml:153]: dev docs is deploying along with docs build and its already there in line 153., if you prefer i can break the dev docs deploy steps too.

Comment (RobPasMue) [reply .github/workflows/ci-build.yml:153]: I see. Can't we keep it all together in the same stage?

Comment (RobPasMue) [reply .github/workflows/ci-build.yml:153]: Split deployment and indexing into separate stages. Have them run after doc build

Comment (Revathyvenugopal162) [reply .github/workflows/ci-build.yml:153]: Ok, i will split it.

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**